app/models/mixins/relationship_mixin.rb

```diff
@@ -769,7 +769,12 @@ def puts_relationship_tree
 
   # this allows us to call similarly named ancestry or mixin methods
   def call_ancestry_method(method_name, *args)
-    bound_method = ((@ancestry_method ||= {})[method_name] ||= ::Ancestry::InstanceMethods.instance_method(method_name)).bind(self)
+    bound_method = if Ancestry::MaterializedPath::InstanceMethods.public_instance_methods.include?(method_name)
+                     ((@ancestry_method ||= {})[method_name] ||= ::Ancestry::MaterializedPath::InstanceMethods.instance_method(method_name)).bind(self)
+                   else
+                     ((@ancestry_method ||= {})[method_name] ||= ::Ancestry::InstanceMethods.instance_method(method_name)).bind(self)
+                   end
+
```

Comment:
considering how long we want this code to live, sorry to say that this LGTM
(I know you were hoping I would have some ultra slick alternative)